internal/services/databricks/databricks_virtual_network_peerings_resource.go

```diff
@@ -0,0 +1,324 @@
+package databricks
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/databricks/2023-02-01/vnetpeering"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/databricks/validate"
+	networkValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/network/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+// peerMutex is used to prevent multiple Peering resources being created, updated
+// or deleted at the same time
+var peerMutex = &sync.Mutex{}
```

Comment:
we have unified mutex's within the provider: [XXX](https://github.com/hashicorp/terraform-provider-azurerm/tree/main/internal/locks)

can we update this to use that?

Reply:
Sure, just did that... running tests now... 🙂

Reply:
Hmmm... 😞 ... looks like my test won't run for another 10 hours as there are 385 tests in front of the test run and I can't skip in the line... running tests locally as a stop gap measure, to see if they pass or not...

Reply:
Manual local tests pass...

<img width="541" alt="image" src="https://user-images.githubusercontent.com/20408400/225260267-2fd24dd4-6401-4d1b-bfd5-441e21079b92.png">